cezo_fl/server.py

```diff
@@ -231,8 +231,14 @@ def train_one_step(self, iteration: int) -> tuple[float, float]:
 
         # Step 3: server-side aggregation
         avg_grad_scalar: list[torch.Tensor] = []
+        # method 1: using average
+        # for each_client_update in zip(*local_grad_scalar_list):
+        #     avg_grad_scalar.append(sum(each_client_update).div_(self.num_sample_clients))
+        # method 2: using median
         for each_client_update in zip(*local_grad_scalar_list):
-            avg_grad_scalar.append(sum(each_client_update).div_(self.num_sample_clients))
+            each_client_tensor = torch.stack(each_client_update)
+            median = torch.median(each_client_tensor, dim=0).values
+            avg_grad_scalar.append(median)
```

Comment:
Should we use dispatch pattern so that we can easily switch and extend the aggregation method?
```python
for each_client_update in zip(*local_grad_scalar_list):
    match self.agg_style:
        case "average":
            avg_grad_scalar.append(avg_tensor(each_client_update))
        case "median":
             avg_grad_scalar.append(median_tensor(each_client_update))
```
We can define `avg_tensor` and `median_tensor` func in util file for future Byzantine problem extension

Reply:
choose to use if else statement for now, can improve using the match/case pattern later